examples/asr/transcribe_speech.py

```diff
@@ -400,9 +416,14 @@ def autocast(dtype=None):
         logging.info(f"Finished transcribing {len(filepaths)} files !")
     logging.info(f"Writing transcriptions into file: {cfg.output_filename}")
 
-    # if transcriptions form a tuple of (best_hypotheses, all_hypotheses), extract just best hypothesis
+    # if transcriptions form a tuple of (best_hypotheses, all_hypotheses)
     if type(transcriptions) == tuple and len(transcriptions) == 2:
-        transcriptions = transcriptions[0]
+        if isinstance(transcriptions[1][0], list):
```

Comment:
Why would this be a list? Make sure to return an NBestHypothesis object. Inside it, the list of all hypotheses is packed.

Reply:
NBestHypothesis object should be returned by both the MultiTaskDecoding so the list is preserved inside of it.

Reply:
Don't check like this. See if the flag is set or not, then extract out the second item in the top level. Almost always, the second item is None rather than the beam values.

Reply:
done